packages/web/src/components/modals/TransactionInfoModals/TransactionStatusSendModal/TransactionStatusSendModal.tsx

```diff
@@ -0,0 +1,161 @@
+const CHECK_TRANSACTION_INTERVAL = 3000;
+
+export const TransactionStatusSendModal: FunctionComponent<
+  ModalPropsType<string | null> & TransactionStatusModalProps
+> = ({ type, action, params, sendState, userFreeFeeLimits, networkFees, close }) => {
+  const { provider } = useWallet();
+
+  const { network } = useConnectionContext();
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [signature, setSignature] = useState<string | null>(null);
+  const transaction = useTransaction(signature as string);
+  const [transactionError, setTransactionError] = useState(
+    transaction?.raw?.meta?.err ? DEFAULT_TRANSACTION_ERROR : '',
+  );
+
+  const executeAction = async () => {
+    try {
+      setIsExecuting(true);
+
+      switch (type) {
+        case 'send': {
+          const resultSignature = await action();
+          setSignature(resultSignature);
+
+          transferNotification({
+            header: 'Sent',
+            text: `- ${params.amount.formatUnits()}`,
+            symbol: params.amount.token.symbol,
+          });
+
+          break;
+        }
+        default:
+          throw new Error('Wrong type');
+      }
+    } catch (error) {
+      // setTransactionError((error as Error).message);
+      setIsExecuting(false);
+
+      if (type === 'send') {
+        ToastManager.error(type, (error as Error).message);
+      }
+    }
+  };
+
+  useEffect(() => {
+    void executeAction();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const mount = async () => {
+      if (!signature) {
+        return;
+      }
+
+      try {
+        const trx = await provider.connection.getTransaction(signature, {
+          commitment: 'confirmed',
+        });
+        if (trx) {
+          if (trx.meta?.err) {
+            setTransactionError(DEFAULT_TRANSACTION_ERROR);
+          } else if (transactionError) {
+            setTransactionError('');
+          }
+        } else {
+          setTimeout(mount, CHECK_TRANSACTION_INTERVAL);
+        }
+      } catch (error) {
+        // setTransactionError((error as Error).message);
+        ToastManager.error((error as Error).message);
+      } finally {
+        setIsExecuting(false);
+      }
+    };
+
+    void mount();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [signature]);
+
+  const isProcessing = (!signature || !transaction?.key) && !transactionError;
+  const isSuccess = Boolean(signature && transaction?.key && !transactionError);
+  const isError = Boolean(transactionError);
+
+  const shortAddress = sendState?.destinationAddress.replace(
+    sendState?.destinationAddress.substring(
+      ADDRESS_CHARS_SHOW,
+      sendState?.destinationAddress.length - ADDRESS_CHARS_SHOW,
+    ),
+    '...',
+  );
+
+  const handleCloseClick = () => {
+    trackEvent('send_close_click', { transactionConfirmed: !isExecuting });
+
+    close(signature);
+  };
+
+  return (
+    <Wrapper>
```

Comment:
Test this modal on mobile too. I think you loose ability to close modal by swipe down, because don't pass `close` func and made it optional.
<img width="460" alt="image" src="https://user-images.githubusercontent.com/656531/165282964-766ade39-7303-47ba-be0e-595d9d135683.png">